folium/vector_layers.py

```diff
@@ -159,16 +159,27 @@ class PolyLine(Marker):
                     {{this.location}},
                     {{ this.options }}
                     )
-                    {% if this.tooltip %}.bindTooltip("{{this.tooltip.__str__()}}"){% endif %}
+                    {% if this.tooltip %}.bindTooltip("{{ this.tooltip }}"){% endif %}
                     .addTo({{this._parent.get_name()}});
             {% endmacro %}
             """)  # noqa
 
     def __init__(self, locations, popup=None, tooltip=None, **kwargs):
         super(PolyLine, self).__init__(location=locations, popup=popup)
         self._name = 'PolyLine'
-        self.tooltip = tooltip
-
+        if tooltip:
+            if isinstance(tooltip, Tooltip):
+                assert not all((tooltip.text, tooltip.fields)), "Only text " \
+                                                                "may be " \
+                                                                "passed to a " \
+                                                                "Polyline " \
+                                                                "Tooltip."
+                self.add_child(tooltip, name=tooltip._name)
+            elif isinstance(tooltip, str):
+                self.tooltip = tooltip.__str__()
```

Comment:
Use a Tooltip object here, that's why we have it now.

`self.add_child(Tooltip(tooltip))`

Reply:
Smart, eliminate the redundancy and funnel everything through the `Tooltip` object and template. This makes it way more maintainable. Thanks for this suggestion.